actions/import.go

```diff
@@ -194,6 +194,81 @@ func (i *Import) importFile(fileName string) error {
 		return errors.Wrap(err, "parse template type")
 	}
 
+	switch templateType {
+	default:
+		return errors.Errorf("unable to handle components of type %s", templateType)
+	case prototype.YAML:
+		return i.createYAML(fileName, base, ext)
+	case prototype.JSON, prototype.Jsonnet:
+		return i.createComponent(fileName, base, ext, templateType)
+	}
+}
+
+func (i *Import) createYAML(fileName, base, ext string) error {
+	readers, err := utilyaml.Decode(i.app.Fs(), fileName)
+	if err != nil {
+		return err
+	}
+
+	for _, r := range readers {
+		data, err := ioutil.ReadAll(r)
+		if err != nil {
+			return err
+		}
+
+		dataReader := bytes.NewReader(data)
+		ts, props, err := schema.ImportYaml(dataReader)
+		if err != nil {
+			if err == schema.ErrEmptyYAML {
+				continue
+			}
+			return err
+		}
+
+		val, err := props.Value([]string{"metadata", "name"})
+		if err != nil {
+			return err
+		}
+
+		name, ok := val.(string)
+		if !ok {
+			return errors.Errorf("unable to find metadata name of object in %s", fileName)
+		}
+
+		componentName := fmt.Sprintf("%s-%s", strings.ToLower(ts.Kind()), name)
```

Comment:
unfortunately, Kind + Name is not unique identifier :( Look at Voyager Ingress for example: https://github.com/appscode/voyager/blob/master/docs/examples/ingress/types/loadbalancer/ing.yaml

it clashes with builtin Ingress.

Kind + Name + Apiversion would be unique, but at least builtin  resources (Deployment, Job, etc) somehow automagically coalesce into same resource, so for them Kind + Name  is unique combination.

Reply:
This is a good catch, I took a look at the built in resources and there wasn't an issue. I will follow up with a change that will handle custom resources as well.

Reply:
It's now kind-name-random.

Adding apiVersion could have made it unbearable. If this doesn't work out, We'll revisit this.